scheduler/replica_scheduler.go

```diff
@@ -261,13 +301,15 @@ func (rcs *ReplicaScheduler) filterNodeDisksForReplica(node *longhorn.Node, disk
 		}
 		if !diskFound {
 			logrus.Errorf("Cannot find the spec or the status for disk %v when scheduling replica", diskUUID)
+			verboseErrors.AppendError(longhorn.SchedulerErrorDiskNotFound)
 			continue
 		}
 		if requireSchedulingCheck {
 			info, err := rcs.GetDiskSchedulingInfo(diskSpec, diskStatus)
 			if err != nil {
 				logrus.Errorf("Fail to get settings when scheduling replica: %v", err)
-				return preferredDisk
+				verboseErrors.AppendError(longhorn.SchedulerErrorSettingIOError)
```

Comment:
Suggest changing the error much meaningful like `longhorn.SchedulerErrorSchedulingSettingsGetError`